`playing` event isn't fired as early as it should

jyavenard opened the pull request:
#### 84eb2324cde995e9b53823137cb06a1af4aaf0d6
<pre>
`playing` event isn&apos;t fired as early as it should
<a href="https://bugs.webkit.org/show_bug.cgi?id=254256">https://bugs.webkit.org/show_bug.cgi?id=254256</a>
rdar://107041118

Reviewed by Youenn Fablet.

Notify the HTMLMediaElement that the media time has changed on the first
change, bypassing the optimisation introduced by webkit.org/b/225396

Code will be exercised by media/media-source/media-source-end-of-stream-buffered.html
once the MockMediaPlayerMediaSource will be running in the GPU process
(webkit.org/b/225367)
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.cpp:
(WebKit::MediaPlayerPrivateRemote::currentTimeChanged):

Canonical link: <a href="https://commits.webkit.org/261959@main">https://commits.webkit.org/261959@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3646e7f9f1838478df93e18183549ec5b424e20a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/108 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/111 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/115 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/109 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/94 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/107 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/105 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/132 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/109 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/88 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/100 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/118 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/112 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/96 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/92 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/90 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/103 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/106 "10 api tests failed or timed out") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/99 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/86 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/85 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/100 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/103 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.cpp:
```diff
@@ -423,7 +424,7 @@ void MediaPlayerPrivateRemote::currentTimeChanged(const MediaTime& mediaTime, co
     m_cachedMediaTimeQueryTime = queryTime;
 
-    if (reverseJump) {
+    if (reverseJump || (!oldCachedTime && oldCachedTime != mediaTime)) {
```
Should it be  `(oldCachedTime && oldCachedTime != mediaTime)`?

jyavenard replied:
no, it needs to be done only when the previous time was 0 (eg. playback didn't start)

I'm actually guessing it should be done if we go from `m_timeIsProgressing = false` to `m_timeIsProgressing = true` but as this code has caused lots of regression in the past, I prefer to take the safe approach and narrow the scope

youennf replied:
Then let's write it as: `(!oldCachedTime && !!mediaTime)`?

jyavenard committed:
`playing` event isn't fired as early as it should
https://bugs.webkit.org/show_bug.cgi?id=254256
rdar://107041118

Reviewed by Youenn Fablet.

Notify the HTMLMediaElement that the media time has changed on the first
change, bypassing the optimisation introduced by webkit.org/b/225396

Code will be exercised by media/media-source/media-source-end-of-stream-buffered.html
once the MockMediaPlayerMediaSource will be running in the GPU process
(webkit.org/b/225367)
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.cpp:
(WebKit::MediaPlayerPrivateRemote::currentTimeChanged):

Canonical link: https://commits.webkit.org/261959@main
